Patch-Source: https://gitlab.gnome.org/World/Chatty/-/merge_requests/1471
---
From c3e46df43b71b38c3c32f35b994fe46aeb18de6e Mon Sep 17 00:00:00 2001
From: Evangelos Ribeiro Tzaras <devrtz@fortysixandtwo.eu>
Date: Fri, 18 Apr 2025 09:48:31 +0200
Subject: [PATCH] treewide: Fix xdg-activation

xdg-activation was broken due to a mismatch of app id, desktop/dbus
service file.

Fixes 70d1ba4ebaa43937dd304b733ceed9c8c0296900
---
 data/meson.build                                              | 4 ++--
 data/{sm.puri.Chatty.service.in => sm.puri.chatty.service.in} | 0
 meson.build                                                   | 1 +
 src/chatty-application.c                                      | 2 +-
 4 files changed, 4 insertions(+), 3 deletions(-)
 rename data/{sm.puri.Chatty.service.in => sm.puri.chatty.service.in} (100%)

diff --git a/data/meson.build b/data/meson.build
index 1e75215e..0672e6a9 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -71,9 +71,9 @@ endif
 
 # DBus service
 service_data = configuration_data()
-service_data.set('appid', app_id)
+service_data.set('appid', app_id.to_lower())
 service_data.set('bindir', get_option('prefix') / get_option('bindir'))
-service_file = '@0@.service'.format(app_id)
+service_file = '@0@.service'.format(app_id.to_lower())
 configure_file(
   input : service_file + '.in',
   output : service_file,
diff --git a/data/sm.puri.Chatty.service.in b/data/sm.puri.chatty.service.in
similarity index 100%
rename from data/sm.puri.Chatty.service.in
rename to data/sm.puri.chatty.service.in
diff --git a/meson.build b/meson.build
index 8f66af15..274d65eb 100644
--- a/meson.build
+++ b/meson.build
@@ -41,6 +41,7 @@ config_h.set_quoted('PACKAGE_VERSION', meson.project_version())
 config_h.set_quoted('BUILD_DIR', meson.current_build_dir())
 config_h.set_quoted('SOURCE_DIR', meson.current_source_dir())
 config_h.set_quoted('CHATTY_APP_ID', app_id)
+config_h.set_quoted('LOWERCASE_APP_ID', app_id.to_lower())
 
 configure_file(
   output: 'config.h',
diff --git a/src/chatty-application.c b/src/chatty-application.c
index 2505d9b9..f37f6546 100644
--- a/src/chatty-application.c
+++ b/src/chatty-application.c
@@ -484,7 +484,7 @@ ChattyApplication *
 chatty_application_new (void)
 {
   return g_object_new (CHATTY_TYPE_APPLICATION,
-                       "application-id", CHATTY_APP_ID,
+                       "application-id", LOWERCASE_APP_ID,
                        "flags", G_APPLICATION_HANDLES_COMMAND_LINE,
                        "register-session", TRUE,
                        NULL);
-- 
GitLab

